crates/sui-data-ingestion/src/reader.rs

```diff
@@ -42,12 +46,14 @@ impl LocalReader {
             }
         }
         files.sort();
+        debug!(
+            "local reader: current checkpoint number is {}. Unprocessed local files are {:?}",
+            current_checkpoint_number, files
+        );
         for (idx, (sequence_number, filename)) in files.iter().enumerate() {
-            assert_eq!(
-                *sequence_number,
-                current_checkpoint_number + idx as u64,
-                "checkpoint sequence should not have any gaps"
-            );
+            if current_checkpoint_number + idx as u64 != *sequence_number {
+                return Err(anyhow!("checkpoint sequence should not have any gaps"));
+            }
             let checkpoint = Blob::from_bytes::<CheckpointData>(&fs::read(filename)?)?;
```

Comment:
there's a slight race when a file is being opened for writing by a script process, but hasn't been flushed yet. This can lead to empty bytes. The error is transient in nature